### PR TITLE
Mttl 1159

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/jira/software/projects/MTTL/boards/47?selectedIssue=MTTL-1159)

## Describe this PR

Raise Tenure Created Event

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: TenureInformationApi.Tests/AwsIntegrationTests.cs [hunk 1]
```diff
@@ -49,6 +58,20 @@ private static void EnsureEnvVarConfigured(string name, string defaultValue)
...
 
+        private void CreateSnsTopic()
+        {
+            var snsAttrs = new Dictionary<string, string>();
+            snsAttrs.Add("fifo_topic", "true");
+            snsAttrs.Add("content_based_deduplication", "true");
+
+            var response = SimpleNotificationService.CreateTopicAsync(new CreateTopicRequest
+            {
+                Name = "tenurecreated",
```
Comment (mikemoody-amido) [inline TenureInformationApi.Tests/AwsIntegrationTests.cs:69]: The name should just be tenure.

File: TenureInformationApi.Tests/AwsIntegrationTests.cs [hunk 2]
```diff
@@ -49,6 +58,20 @@ private static void EnsureEnvVarConfigured(string name, string defaultValue)
...
+                Attributes = snsAttrs
+            }).Result;
+
+            Environment.SetEnvironmentVariable("NEW_TENURE_SNS_ARN", response.TopicArn);
```
Comment (mikemoody-amido) [inline TenureInformationApi.Tests/AwsIntegrationTests.cs:73]: You should probably remove the NEW_ from this env var name.

File: TenureInformationApi.Tests/V1/E2ETests/Fixtures/TenureFixture.cs [hunk 3]
```diff
@@ -63,10 +71,26 @@ public void GivenNewTenureRequestWithValidationErrors()
...
+            snsAttrs.Add("content_based_deduplication", "true");
+
+            var response = _amazonSimpleNotificationService.CreateTopicAsync(new CreateTopicRequest
+            {
+                Name = "tenurecreated",
+                Attributes = snsAttrs
+            }).Result;
+
+            Environment.SetEnvironmentVariable("NEW_TENURE_SNS_ARN", response.TopicArn);
+        }
```
Comment (mikemoody-amido) [inline TenureInformationApi.Tests/V1/E2ETests/Fixtures/TenureFixture.cs:92]: You don't need this here.
The test story class already has the `Aws collection` set on it meaning that the implementation within the `AwsIntegrationTests` class will have already been run.

Comment (humulla) [reply TenureInformationApi.Tests/V1/E2ETests/Fixtures/TenureFixture.cs:92]: But the method in integration tests is currently private, also we want to create the SNS topic when the request object is created

File: .localstack/.localstack [hunk 4]
```diff
@@ -0,0 +1 @@
+{"machine_id": "5ec5ee7a"}
```
[the remark this answers is not in the document]
Comment (humulla) [reply .localstack/.localstack:1]: I'm confused, I didn't create this file it was auto-created when adding the local stack stuff in the docker-compose

File: TenureInformationApi/Startup.cs [hunk 5]
```diff
@@ -129,27 +132,36 @@ public void ConfigureServices(IServiceCollection services)
                 var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
                 if (File.Exists(xmlPath))
                     c.IncludeXmlComments(xmlPath);
+                services.Configure<AwsConfiguration>(options => Configuration.GetSection("AWS").Bind(options));
```
[the remark this answers is not in the document]
Comment (humulla) [reply TenureInformationApi/Startup.cs:135]: I'm not sure, I thought we needed some configuration stuff but it doesn't look like we do, I'm going to remove it

File: TenureInformationApi.Tests/V1/E2ETests/Steps/PostTenureSteps.cs [hunk 6]
```diff
@@ -55,7 +55,7 @@ public async Task WhenCreateTenureApiIsCalled(CreateTenureRequestObject requestO
 
         public async Task ThenTheTenureDetailsAreReturnedAndIdIsNotEmpty(TenureFixture tenureFixture)
         {
-            _lastResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+            //_lastResponse.StatusCode.Should().Be(HttpStatusCode.Created);
```
[the remark this answers is not in the document]
Comment (humulla) [reply TenureInformationApi.Tests/V1/E2ETests/Steps/PostTenureSteps.cs:58]: Sorry commented by mistake, will uncomment

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**